src/parser.lalrpop

```diff
@@ -74,7 +74,7 @@ BuiltInType: HLLString = {
 }
 
 FuncDecl: Box<FuncDecl> = {
-	"fn" <n: NameDecl> "(" <a: Comma<FuncDeclArg>> ")" "{" <b: Stmt+> "}" => Box::new(FuncDecl { class_name: None, name: n, args: a, body: b, annotations: Annotations::new() }),
+	"fn" <n: NameDecl> "(" <a: Comma<FuncDeclArg>> ")" "{" <b: Stmt*> "}" => Box::new(FuncDecl::new(n, a, b)),
```

Comment:
We could allow or force a final `;` instead of an empty `{ }`. I think it would be cleaner to deny an empty body, similarly to a function signature declaration in C or Rust.

Reply:
I've gone back and forth on that a few times.  I think an argument in favor of the empty `{}` syntax is that that's the expected syntax for implementing a type with no specific rules:

```
resource foo inherits some_parent {}
```

Doing it this way creates a clean "declarations are always blocks" and "statements end in semicolon" distinction.  I see the "like C and rust" argument, so I'm open to pushback, but I do think this way is more internally consistent.  I think declaring virtual functions with a semicolon, becomes an exception that needs to be memorized.  It's a logical an obvious exception, but an exception none the less.